openssh: allow multiple config files

pgwipeout opened the pull request:
The default openssh init file only permits a single sshd instance.
However it is common to implement multiple servers with different
configuration files.
This allows us to define the configuration file sshd uses, permitting
multiple sshd instances.

Signed-off-by: Peter Geis <pgwipeout@gmail.com>

Maintainer: me / @\<tripolar> (find it by checking history of the package Makefile)
Compile tested: (aarch64 armvirt OpenWrt 19.07.1)
Run tested: (aarch64 armvirt OpenWrt 19.07.1)

Description:
Permit multiple sshd instances by defining the config file used.

neheb commented on net/openssh/files/sshd.init:
```diff
@@ -6,27 +6,28 @@ STOP=50
-		key=/etc/ssh/ssh_host_${type}_key
-		[ ! -f $key ] && {
+		type=`echo $keys | grep -o 'rsa\|ecdsa\|ed25519'`
+		[ ! -f $keys ] && {
```
Should be changed to [ -f ] ||

neheb commented on net/openssh/files/sshd.init:
```diff
@@ -6,27 +6,28 @@ STOP=50
 		# check for keys
-		key=/etc/ssh/ssh_host_${type}_key
-		[ ! -f $key ] && {
+		type=`echo $keys | grep -o 'rsa\|ecdsa\|ed25519'`
```
Outdated syntax. $() is better.

neheb commented on net/openssh/files/sshd.init:
```diff
@@ -6,27 +6,28 @@ STOP=50
 
 start_service() {
-	for type in rsa ecdsa ed25519
+	for keys in `awk '/^HostKey / { print $2 }' $CONF`
```
$()

neheb commented:
Down in shutdown(), == should be replaced with =. == is a bashism.

Actually, shutdown() should probably be replaced with stop_service(). I wonder if it even gets used...

pgwipeout commented:
> Down in shutdown(), == should be replaced with =. == is a bashism.
> 
> Actually, shutdown() should probably be replaced with stop_service(). I wonder if it even gets used...

While shutdown appears to be a valid node, this seems entirely too destructive and unnecessary with procd's management.

Using the stock stop() from rc.common seems adequate. 

I am tweaking things though to improve behavior.

pgwipeout commented:
So shutdown is actually necessary to prevent backgrounded tasks from hanging a reboot.
I've fixed this up so now shutdown only kills tasks that belonged to that ssh session, and addressed your concerns with formatting.

neheb commented:
Seems some quoting is missing. Run through shellcheck: https://www.shellcheck.net/

pgwipeout commented:
Done.

pgwipeout commented:
I notice that @tripolar added this functionality under his own commit: 594a884c2d6dbfeaa64705f34f17131bb900e952

Closing the request.
